src/Microsoft.AspNet.Mvc.Core/Internal/Routing/LinkGenerationMatch.cs

```diff
@@ -10,14 +13,42 @@ public struct LinkGenerationMatch
         private readonly bool _isFallbackMatch;
         private readonly AttributeRouteLinkGenerationEntry _entry;
 
-        public LinkGenerationMatch(AttributeRouteLinkGenerationEntry entry, bool isFallbackMatch)
+        public LinkGenerationMatch(
+            AttributeRouteLinkGenerationEntry entry,
+            bool isFallbackMatch,
+            VirtualPathContext pathContext)
         {
             _entry = entry;
             _isFallbackMatch = isFallbackMatch;
+            RouteValues = pathContext.Values;
+            AmbientValues = pathContext.AmbientValues;
         }
 
         public AttributeRouteLinkGenerationEntry Entry { get { return _entry; } }
 
         public bool IsFallbackMatch { get { return _isFallbackMatch; } }
+
+        public IDictionary<string, object> RouteValues { get; }
```

Comment:
May want to consider exposing this and `AmbientValues` as an `IReadOnlyDictionary<string, object>`.